opennms-base-assembly/src/main/filtered/bin/_lib.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+__onms_is_array() {
+	__onms_is_array_name="$1"; shift
+	__onms_is_array_declaration="$(declare -p "${__onms_is_array_name}" 2>/dev/null)"
+	[ "$(echo "${__onms_is_array_declaration}" | grep -c '^declare -a')" -gt 0 ];
+}
+
+__onms_convert_array() {
+	__onms_convert_array_name="$1"; shift
+	__onms_convert_array_declaration="$(declare -p "${__onms_convert_array_name}" 2>/dev/null)"
+	if [ -z "${__onms_convert_array_declaration}" ]; then
+		# variable is not set
+    eval "${__onms_convert_array_name}=()"
+	elif __onms_is_array "${__onms_convert_array_name}"; then
+		# variable is an array
+		:
+	else
+		# variable is not an array
+		__onms_convert_array_existing_value="$(eval echo "\${$__onms_convert_array_name}" | sed -e 's,^[[:space:]]*,,' -e 's,[[:space:]]*$,,')"
+		unset "${__onms_convert_array_name?}"
+    eval "${__onms_convert_array_name}=()"
+		IFS=" " read -r -a "${__onms_convert_array_name?}" <<< "${__onms_convert_array_existing_value}"
```

Comment:
Should we reset `IFS` here since we changed it?

Reply:
It's not changed when you use it at the beginning of a command like that, it only affects that line.